plugin-server/src/worker/ingestion/group-type-manager.ts

```diff
@@ -34,11 +40,39 @@ export class GroupTypeManager {
         if (groupType in groupTypes) {
             return groupTypes[groupType]
         } else {
-            const response = await this.db.insertGroupType(teamId, groupType, Object.keys(groupTypes).length)
-            if (response !== null) {
+            const [groupTypeIndex, isInsert] = await this.db.insertGroupType(
+                teamId,
+                groupType,
+                Object.keys(groupTypes).length
+            )
+            if (groupTypeIndex !== null) {
                 this.groupTypesCache.delete(teamId)
             }
-            return response
+
+            if (isInsert && groupTypeIndex !== null) {
+                await this.captureGroupTypeInsert(teamId, groupType, groupTypeIndex)
+            }
+            return groupTypeIndex
         }
     }
+
+    private async captureGroupTypeInsert(teamId: TeamId, groupType: string, groupTypeIndex: number) {
+        const team: Team | null = await this.teamManager.fetchTeam(teamId)
+
+        if (!team) {
+            return
+        }
+
+        posthog.identify('plugin-server')
```

Comment:
we can get rid of this by adding `distinct_id` as a prop on the call below

Reply:
Looking at the code that doesn't seem to be correct: https://github.com/PostHog/posthog-js-lite/blob/master/src/index.ts#L54, but might be missing something